kernel/include/kernel_internal.h

```diff
@@ -207,6 +207,27 @@ void z_kernel_map_fixup(void);
 			!__i.key;					\
 			k_spin_unlock(lck, __key), __i.key = 1)
 
+#ifdef CONFIG_PM
```

Comment:
first, your commit message is missing a NOT, second, can we put this in some pm specific private header instead?